src/codecs/tiff.rs

```diff
@@ -338,6 +364,61 @@ impl<R: BufRead + Seek> ImageDecoder for TiffDecoder<R> {
 /// Encoder for tiff images
 pub struct TiffEncoder<W> {
     w: W,
+    comp: Compression,
+}
+
+/// Compression types supported by the TIFF format
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum CompressionType {
+    /// No compression
+    Uncompressed,
+    /// LZW compression
+    Lzw,
+    /// Deflate compression
+    Deflate(TiffDeflateLevel),
+    /// Bit packing compression
+    Packbits,
+}
+
+impl Default for CompressionType {
+    fn default() -> Self {
+        CompressionType::Deflate(Default::default())
```

Comment:
As far as I know, LZW is more common for TIFF files so that should probably be the default.

Reply:
Defaulting LZW would seem to be more a historical accident. If we're going with libtiff as upstream we might want to _also_ look into zstd support: http://libtiff.maptools.org/v4.0.10.html

> Please note that COMPRESSION_ZSTD is self-assigned the id 50000
    by the libtiff project and is not officially registered with Adobe
    since Adobe's registration function is defunct.

Yikes.

We should be making our own comparison in terms of size / speed. Based on the compression test this might favor deflate if `Balanced` achieves comparable size to LZW. Which would not be all that surprising, LZW is the oldest of the bunch.